### ISSUE TITLE
Highlight Word functionality stops working 

### ISSUE THREAD

**Issue by RobertvanderHulst** [VsixCommunity/Community.VisualStudio.Toolkit#385]
The Highlight Word functionality stops working  when the "Preview ToolTip" of the vertical scrollbar is enabled and the mouse moves over the scrollbar, triggering the preview tooltip to be shown.
After closing and reopening the file the highlight word feature works again.

**Comment by madskristensen** [comment]
Is that an issue with VS or the toolkit, you think?

**Comment by RobertvanderHulst** [comment]
I see the same problem in my own extension as well as the pkgdef language service that you wrote. 
The C# project system does not have that problem.

**Comment by RobertvanderHulst** [comment]
Ok, I found the cause of the problem.
The Preview Tooltip creates another textview over the same textbuffer. That also happens when you split the source code editor or when you Peek definition in the same file.
This new textview also calls CreateTagger and receives the existing SameWordHighlighterTagger because they share the same TextBuffer.
When the new textview is closed then the Dispose() method of the SameWordHighlighterTagger() is called, and this unregisters the event handlers for the SameWordHighlighterTagger.

I have added a Counter property to the SameWordHighlighterTagger class. Each time it is "used" the counter is incremented and inside Dispose() the counter is decremented. Only when the counter reaches 0 then the event handlers are unregistered.
This works.
I have also added a HiglightWordTagger to the example project to help test this.
I'll send a pull request for this later.
